pkg/util/liveconfig/hive.go

```diff
@@ -0,0 +1,57 @@
+package liveconfig
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	mgmtcontainerservice "github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-10-01/containerservice"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func parseKubeconfig(credentials []mgmtcontainerservice.CredentialResult) (*rest.Config, error) {
+	res := make([]byte, base64.StdEncoding.DecodedLen(len(*credentials[0].Value)))
+	_, err := base64.StdEncoding.Decode(res, *credentials[0].Value)
+	if err != nil {
+		return nil, err
+	}
+
+	clientconfig, err := clientcmd.NewClientConfigFromBytes(res)
+	if err != nil {
+		return nil, err
+	}
+
+	restConfig, err := clientconfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return restConfig, nil
+}
+
+func (p *prod) HiveRestConfig(ctx context.Context, index int) (*rest.Config, error) {
+	cached, ext := p.cachedCredentials[index]
```

Comment:
Since we know rotation happens every 30 years, this is fine. But in general I would like to see TTL for caching.

Reply:
what does `ext` mean ? Shouldn't it be `exists` or `ok`?

Reply:
yeah, I figured that it would be more complicated than it's worth here. +1 for TTLs for stuff, though.